pkg/apis/build/v1alpha1/build_types.go

```diff
@@ -138,6 +138,11 @@ type BuildSpec struct {
 	// Env contains additional environment variables that should be passed to the build container
 	// +optional
 	Env []corev1.EnvVar `json:"env,omitempty"`
+
+	// Volumes contains volume Overrides of the BuildStrategy volumes in case those are allowed
+	// to be overriden. Must only contain volumes that exist in the corresponding BuildStrategy
+	// +optional
+	Volumes []BuildVolume `json:"volumes,omitempty"`
```

Comment:
`BuildVolume` contains settings that are not applicable for Builds and BuildRuns: `optional` and `overridable`. We will need separate types.

Reply:
That makes sense, I will create separate types then.